File: 백준/11053.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;	

int main() {
	int N;
	cin >> N;
	vector <int> arr(1001);
	vector <int> DP(1001);
	for (int i = 0; i < N; i++) {
		cin >> arr[i];
	}
	int answer = 1;
	DP[0] = 1;
	for (int i = 1; i < N; i++) {
		DP[i] = 1;
		for (int j = 0; j < i; j++) {
			if (arr[i] > arr[j]) {
				DP[i] = max(DP[i], DP[j] + 1);
			}
		}
		answer = max(answer, DP[i]);
	}
	cout << answer;
}

File: 백준/11053.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
const N=Number(input[0]);
const [...arr]=input[1].split(' ').map(Number);
let dp=Array(N).fill(1);
for (let i=1; i<N; i++) {
    for (let j=i-1; j>=0; j--) {
        if (arr[j]<arr[i]) {
            dp[i]=Math.max(dp[i], dp[j]+1);
        }
    }
}
console.log(Math.max.apply(null, dp))
